### PR TITLE
comments and code cleanup

### PR DESCRIPTION
**Pull request by PopGoesTheWza**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -189,7 +205,7 @@ const ts2gas = (source: string, transpileOptions: ts.TranspileOptions = {}) => {
   // ## Exports
   // Exports are transpiled to variables 'exports' and 'module.exports'
 
-  const pjson = require('../package.json');
+  const pjson = require('../package.json');  // ugly hack
```
Comment (grant) [inline src/index.ts]: I don't think this is really ugly, besides the relative require.

File: src/index.ts [hunk 2]
```diff
@@ -121,19 +137,19 @@ const ts2gas = (source: string, transpileOptions: ts.TranspileOptions = {}) => {
...
+  /**  Filter ts.Node which are statement assigning to 'exports["default"]' */
   const exportsDefaultNodeFilter: NodeFilter = (node: ts.Node) =>
     ts.isExpressionStatement(node)
-    && ts.isBinaryExpression(node.expression)
+    && ts.isBinaryExpression(node.expression)  // is it a binary expression
     && ts.isPropertyAccessExpression(node.expression.left)
-    && node.expression.left.expression.hasOwnProperty('escapedText')
-    && (node.expression.left.expression as unknown as {escapedText: string}).escapedText === 'exports'
-    && ts.isIdentifier(node.expression.left.name)
+    && ts.isIdentifier(node.expression.left.expression)  // is it ''exports
```
Comment (grant) [inline src/index.ts]: 'exports'

File: src/index.ts (hunk 1 above)
Comment (PopGoesTheWza) [reply src/index.ts]: it's more because lots of statements stand here and there without e clear structure/design. Like the types and interfaces I added

**Comment by PopGoesTheWza** [comment]
@grant when you next release ts2gas, wouldn't it make sense to also bump clasp with ts2gas dependency updated?

**Comment by grant** [comment]
OK, I can bump clasp.
In general, create issues for what you want to see happen.

**Pull request merged**